numpy/ctypeslib.py

```diff
@@ -509,10 +509,32 @@ def as_array(obj, shape=None):
         """
         Create a numpy array from a ctypes array or POINTER.
 
-        The numpy array shares the memory with the ctypes object.
-
-        The shape parameter must be given if converting from a ctypes POINTER.
-        The shape parameter is ignored if converting from a ctypes array
+        The numpy array shares memory with the ctypes object.
+        
+        Parameters
+        ----------
+        obj : ctypes Pointer or ctypes array
+            Data object with contiguous chunk of memory for numpy.ndarray data
+        shape : int or tuple of int, optional
+            The shape of the returned numpy.ndarray. `shape` must be given if
+            converting from a ctypes POINTER, but is ignored if converting from
+            a ctypes array.
+            
+        Returns
+        -------
+        numpy.ndarray
+            numpy.ndarray sharing memory of input obj
+            
+        Notes
+        -----                        
+        Numpy array stores its data as a void pointer in C. If you use a Pointer,
+        you must use ctypes Single Pointer.
```

Comment:
What is a ctypes Single Pointer? There is no such documented ctype. And why is Pointer capitalized?

Reply:
@charris 
ctypes Single Pointer : i mean Python 'ctypes.POINTER'. 
[ctypes.pointer](https://docs.python.org/ko/3/library/ctypes.html#ctypes.pointer)
the reason why i use single is to emphasize not using double pointer
(because i using double pointer for numpy array data source before and have issue about it [issue](https://github.com/numpy/numpy/issues/21335))

Pointer capitalized : there are no mean using capitalized.
(i just think it can emphasize using pointer, but it might a bad choice. please amend to lower case ㅠㅠ)

Reply:
Sorry for not following up earlier.  I think the best solution w.r.t. the "Single Pointer" (which is indeed maybe not super clear without context), may be to rather create an
```
Examples
--------
```
section, and make an example for a 2-D array?

The example section is missing anyway and I suspect that it would be much clearer.